Home_Work13/HW13.cpp

```diff
@@ -0,0 +1,442 @@
+#define _CRT_SECURE_NO_WARNINGS
+#include <iostream>
+#include <fstream>
+#include <string>
+#include <algorithm>
+#include <ctime>
+
+enum class readParameters //defines which string to read from save file
+{
+	Date,
+	Word,
+	Status
+};
+
+void printMainMenu();
+void printGameStartedMenu(int attempts);
+std::string mainModeErrors(std::string mainModeChoise);
+std::string gameModeErrors(std::string inputString, int wordLength);
+std::string intermidiateWordResult(std::string inputString, std::string guessWord, const int wordLength);
+std::string generateWord(const std::string WordsFile, const int WordsCount, const int wordLengt);
+void unmatchedLettersList(std::string inputString, std::string guessWord, const int wordLength, std::string& usedUnmatchLetters);
+bool checkWinCondition(std::string guessWord, std::string resultAfterGuess, const int wordLength);
+void printNewRoundMenu(std::string inputString, std::string resultAfterGuess, std::string usedUnmatchLetters, int currentAttemptsCounter);
+void printWinMenu(std::string guessWord, const int maxAttempts, int currentAttemptsCounter);
+void printlossMenu(std::string guessWord, std::string mainModeChoise);
+void printError(std::string errors);
+std::string readSavedData(std::string saveDataToFile, readParameters parameterToRead);
+std::string getCurrentDate();
+void writeSaveData(std::string date, std::string word, std::string status, const std::string saveDataToFile);
+
+int main()
+{
+	const int usualWordsCount = 212; //number of words for usual mode, changing this requires changing number of words in data base
```

Comment:
this const can easily be a part of the file itself. In general, someone who may be modifying the file shouldn't worry about some code changes, just need to follow the format. That is, adding 15 more words into file db should work without code changes